adam/visualization/make_scenes.py

```diff
@@ -330,6 +336,14 @@ def main(
                         repositioned_map.name_to_scale[name],
                     )
 
+            if gaze_arrows:
+                for _, handle in scene_elements.situation_object_to_handle.items():
```

Comment:
The issue here is that not every object has the `gazed_at` property. (Annoyingly, in the basic curriculum everything is gazed at. I tried 

```from adam.curriculum.verbs_with_dynamic_prepositions_curriculum import (
     _make_take_with_prepositions as make_curriculum,
)
```
and found a scene where only some of the objects were being looked at. 

So you'll want to use the `SceneElements.property_map` to check which things have the binary property gazed-at.

Reply:
I would suggest changing the key of that `property_map` (line 464) from `ObjectPerception` just to a `str` (using `ObjectPerception.debug_handle` as the key.